scripts/safe_template_linter.py

```diff
@@ -577,6 +497,59 @@ def print_results(self, options, out):
             ), file=out)
 
 
+class SummaryResults(object):
+    """
+    Contains the summary results for all violations.
+    """
+
+    def __init__(self):
+        """
+        Init method.
+        """
+        self.total_violations = 0
+        self.totals_by_rule = dict.fromkeys(
+            [rule.rule_id for rule in Rules.__members__.values()], 0
+        )
+
+    def add_violation(self, violation):
+        """
+        Adds a violation to the summary details.
+
+        Arguments:
+            violation: The violation to add to the summary.
+
+        """
+        self.total_violations += 1
+        self.totals_by_rule[violation.rule.rule_id] += 1
+
+    def print_results(self, options, out):
+        """
+        Prints the results (i.e. violations) in this file.
+
+        Arguments:
+            options: A list of the following options:
+                list_files: True to print only file names, and False to print
+                    all violations.
+                rule_totals: If True include totals by rule.
+            out: output file
+
+        """
+        if options['list_files'] is False:
+            if options['rule_totals']:
+                max_rule_id_len = 0
+                for rule_id in self.totals_by_rule.keys():
+                    max_rule_id_len = max(max_rule_id_len, len(rule_id))
```

Comment:
The loop as a whole can be: `max_rule_id_len = max(len(rule_id) for rule_id in self.totals.by_rule)`